src/sse/sse_butterflies.rs

```diff
@@ -0,0 +1,4246 @@
+    .direction);
+impl<T: FftNum> SseF64Butterfly32<T> {
+    #[inline(always)]
+    pub fn new(direction: FftDirection) -> Self {
+        assert_f64::<T>();
+        let bf8 = SseF64Butterfly8::new(direction);
+        let bf16 = SseF64Butterfly16::new(direction);
+        let rotate90 = if direction == FftDirection::Inverse {
+            Rotate90F64::new(true)
+        } else {
+            Rotate90F64::new(false)
+        };
+        let twiddle1 =
+            unsafe { _mm_loadu_pd(&twiddles::compute_twiddle(1, 32, direction).re as *const f64) };
+        let twiddle2 =
+            unsafe { _mm_loadu_pd(&twiddles::compute_twiddle(2, 32, direction).re as *const f64) };
+        let twiddle3 =
+            unsafe { _mm_loadu_pd(&twiddles::compute_twiddle(3, 32, direction).re as *const f64) };
+        let twiddle4 =
+            unsafe { _mm_loadu_pd(&twiddles::compute_twiddle(4, 32, direction).re as *const f64) };
+        let twiddle5 =
+            unsafe { _mm_loadu_pd(&twiddles::compute_twiddle(5, 32, direction).re as *const f64) };
+        let twiddle6 =
+            unsafe { _mm_loadu_pd(&twiddles::compute_twiddle(6, 32, direction).re as *const f64) };
+        let twiddle7 =
+            unsafe { _mm_loadu_pd(&twiddles::compute_twiddle(7, 32, direction).re as *const f64) };
+        let twiddle1c = unsafe {
+            _mm_loadu_pd(&twiddles::compute_twiddle(1, 32, direction).conj().re as *const f64)
+        };
+        let twiddle2c = unsafe {
+            _mm_loadu_pd(&twiddles::compute_twiddle(2, 32, direction).conj().re as *const f64)
+        };
+        let twiddle3c = unsafe {
+            _mm_loadu_pd(&twiddles::compute_twiddle(3, 32, direction).conj().re as *const f64)
+        };
+        let twiddle4c = unsafe {
+            _mm_loadu_pd(&twiddles::compute_twiddle(4, 32, direction).conj().re as *const f64)
+        };
+        let twiddle5c = unsafe {
+            _mm_loadu_pd(&twiddles::compute_twiddle(5, 32, direction).conj().re as *const f64)
+        };
+        let twiddle6c = unsafe {
+            _mm_loadu_pd(&twiddles::compute_twiddle(6, 32, direction).conj().re as *const f64)
+        };
+        let twiddle7c = unsafe {
+            _mm_loadu_pd(&twiddles::compute_twiddle(7, 32, direction).conj().re as *const f64)
+        };
+
+        Self {
+            direction,
+            bf8,
+            bf16,
+            rotate90,
+            twiddle1,
+            twiddle2,
+            twiddle3,
+            twiddle4,
+            twiddle5,
+            twiddle6,
+            twiddle7,
+            twiddle1c,
+            twiddle2c,
+            twiddle3c,
+            twiddle4c,
+            twiddle5c,
+            twiddle6c,
+            twiddle7c,
+        }
+    }
+
+    #[inline(always)]
+    unsafe fn perform_fft_contiguous(
+        &self,
+        input: RawSlice<Complex<T>>,
+        output: RawSliceMut<Complex<T>>,
+    ) {
+        let in0 = _mm_loadu_pd(input.as_ptr() as *const f64);
```

Comment:
Does it hurt code generation to use an array literal here? IE, 

```rust
let in = [
    _mm_loadu_pd(input.as_ptr() as *const f64),
    _mm_loadu_pd(input.as_ptr().add(1) as *const f64),
    _mm_loadu_pd(input.as_ptr().add(2) as *const f64),
   ...
```
If so, that might be a way to reduce the amount of repetition here.

Reply:
That probably won't hurt, and gets shorter. Will do!

Reply:
I tried this, and the speed is the same. I also made a macro to build the array with much less typing.